File: sol/2870.cpp
#include <unordered_map>

struct Solution {
    int minOperations(const vector<int>& nums) {
        unordered_map<int, int> c;
        for (int n : nums){
            ++c[n];
        }
        int res = 0;
        for (auto it = c.begin(); it != c.end(); ++it){
            if (it->second == 1){
                return -1;
            }
            res += (it->second + 2) / 3;
        }
        return res;
    }
};

File: sol/2870/2870.py
from collections import Counter

class Solution:
    def minOperations(self, nums: List[int]) -> int:
        c = Counter(nums)
        res = 0
        for v in c.values():
            if v == 1:
                return -1
            res += (v + 2) // 3
        return res
